indexer/services/comlink/src/controllers/api/v4/social-trading-controller.ts

```diff
@@ -0,0 +1,109 @@
+import { stats } from '@dydxprotocol-indexer/base';
+import {
+  SubaccountFromDatabase,
+  SubaccountTable,
+  SubaccountUsernamesFromDatabase,
+  SubaccountUsernamesTable,
+} from '@dydxprotocol-indexer/postgres';
+import express from 'express';
+import {
+  checkSchema,
+  matchedData,
+} from 'express-validator';
+import {
+  Controller, Get, Query, Route,
+} from 'tsoa';
+
+import { getReqRateLimiter } from '../../../caches/rate-limiters';
+import config from '../../../config';
+import { NotFoundError } from '../../../lib/errors';
+import { handleControllerError, checkIfValidDydxAddress } from '../../../lib/helpers';
+import { rateLimiterMiddleware } from '../../../lib/rate-limit';
+import { handleValidationErrors } from '../../../request-helpers/error-handler';
+import ExportResponseCodeStats from '../../../request-helpers/export-response-code-stats';
+import { subaccountInfoToTraderSearchResponse } from '../../../request-helpers/request-transformer';
+import { TraderSearchRequest, TraderSearchResponse } from '../../../types';
+
+const router: express.Router = express.Router();
+const controllerName: string = 'social-trading-controller';
+
+@Route('trader')
+class SocialTradingController extends Controller {
+  @Get('/search')
+  async searchTrader(
+    @Query() searchParam: string,
+  ): Promise<TraderSearchResponse> {
+    if (checkIfValidDydxAddress(searchParam)) {
+
+      const subaccounts: SubaccountFromDatabase[] = await
+      SubaccountTable.findAll({
+        address: searchParam,
+        subaccountNumber: 0,
+      }, [], {});
+      const subaccount: SubaccountFromDatabase = subaccounts[0];
+
+      if (!subaccount) {
+        throw new NotFoundError(`Subaccount not found:${searchParam}`);
+      }
+
+      const subaccountUsernames: SubaccountUsernamesFromDatabase[] = await
+      SubaccountUsernamesTable.findAll({
+        subaccountId: [subaccount.id],
+      }, [], {});
+
+      return subaccountInfoToTraderSearchResponse(subaccount, subaccountUsernames[0]);
+    }
+
+    const subaccountUsername: SubaccountUsernamesFromDatabase | undefined = await
+    SubaccountUsernamesTable.findByUsername(searchParam);
+
+    if (!subaccountUsername) {
+      throw new NotFoundError(`Subaccount not found:${searchParam}`);
+    }
+
+    const subaccount: SubaccountFromDatabase | undefined = await
+    SubaccountTable.findById(subaccountUsername.subaccountId);
+    if (!subaccount) {
+      throw new NotFoundError(`Subaccount not found:${subaccountUsername.subaccountId}`);
```

Comment:
This is a foreign key right? So this should technically be impossible? Could you add that in comments

Reply:
I agree, this should be impossible. Better to remove frankly